subsys/mgmt/mcumgr/grp/img_mgmt/Kconfig

```diff
@@ -153,6 +153,38 @@ config MCUMGR_GRP_IMG_MUTEX
 	  can be used by applications to reset the image management state (useful if there are
 	  multiple ways that firmware updates can be loaded).
 
+choice MCUMGR_GRP_IMG_TOO_LARGE_CHECK
+	prompt "Image size check overhead"
+	default MCUMGR_GRP_IMG_TOO_LARGE_DISABLED
+	help
+	  MCUboot images should be limited to the maximum size that the bootloader can swap, in
+	  order to know this size, additional information is needed from the MCUboot
+	  configuration, otherwise an image can be uploaded that is too large for the bootloader
+	  to swap, this selects which method to use.
+
+	  Note: setting this to a non-disabled option will prevent uploading of padded and
+	  confirmed images, if support for that is required then this feature should be left as
+	  disabled.
```

Comment:
Would it be possible for us to develop MCUboot/bootutil function that would check the image for valid footer?

Reply:
Not possible, well, depending on what you mean by valid footer. Only MCUboot can confirm the image properly due to possibly being encrypted, an application could do some minor validation but nothing more than that

Reply:
Actually you can use imgtool to pad, confirm and encrypt image. 
But yes, I guess you can not check the footer if the MCUboot only knows the key.

Reply:
Also can't check that here because to check that you need to receive the whole image

Reply:
The image slot trailer, the one that stores swap bits, etc, is not encrypted. actually.

Reply:
Not encrypted no but only mcuboot can verify that it is a valid image or supported on that device, the application can't know anything more than the basic validity of parts of the image